src/cai_lab_to_nwb/zaki_2024/interfaces/miniscope_imaging_interface.py

```diff
@@ -195,38 +191,36 @@ def __init__(self, folder_path: DirectoryPath):
         from ndx_miniscope.utils import get_recording_start_times, read_miniscope_config
 
         super().__init__(folder_path=folder_path)
-        
-        self.miniscope_folder = Path(folder_path)        
+
+        self.miniscope_folder = Path(folder_path)
         # This contains the general metadata and might contain behavioral videos
-        self.session_folder = self.miniscope_folder.parent  
+        self.session_folder = self.miniscope_folder.parent
 
         self._miniscope_config = read_miniscope_config(folder_path=self.miniscope_folder)
-        
+
         # use the frame rate of the json configuration to set the metadata
         frame_rate_string = self._miniscope_config["frameRate"]
         # frame_rate_string look like "30.0FPS", extract the float part
         self._metadata_frame_rate = float(frame_rate_string.split("FPS")[0])
-        
-        
+
         self.photon_series_type = "OnePhotonSeries"
 
     def _get_session_start_time(self):
-
-        general_metadata_json = self.session_folder/ "metaData.json"
+        general_metadata_json = self.session_folder / "metaData.json"
         assert general_metadata_json.exists(), f"General metadata json not found in {self.session_folder}"
 
         ## Read metadata
         with open(general_metadata_json) as f:
             general_metadata = json.load(f)
 
         session_start_time = datetime.datetime(
-            year=general_metadata["year"],
-            month=general_metadata["month"],
-            day=general_metadata["day"],
-            hour=general_metadata["hour"],
-            minute=general_metadata["minute"],
-            second=general_metadata["second"],
-            microsecond=general_metadata["msec"] * 1000,  # Convert milliseconds to microseconds
+            year=general_metadata["recordingStartTime"]["year"],
```

Comment:
This needs to be handled, some files are not nested like this. Can you add code to take this into account? 

In other words, the code that I wrote works for some experiments and yours work for some others. We should handle both cases.

Reply:
You right! I forgot to ask you about this. I would probably just remove the option of setting the session_starting_time automatically

Reply:
I have only seen these two options, if that is the case I think we should handle it. Check if the dictionary has an attribute `recordingStartTime`, if it does not, then check if is in the top layer, otherwise return None.

Reply:
Ah, I saw you did. Marking this as resolved.